src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -143,23 +146,82 @@ private static Assembly LoadCore(string fullPath)
             }
 
             /// <summary>
-            /// Handles the <see cref="AppDomain.AssemblyResolve"/> event when the requesting
-            /// assembly is one that we've loaded.
-            /// 
-            /// We assume that an assembly's dependencies can be found next to it in the file
-            /// system.
+            /// Handles the <see cref="AppDomain.AssemblyResolve"/> event.
             /// </summary>
             private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
             {
                 if (args.RequestingAssembly == null)
                 {
-                    // We don't know who is requesting the load; don't try to satisfy the request.
+                    return ResolveForUnknownRequestor(args.Name);
+                }
+                else
+                {
+                    return ResolveForKnownRequestor(args.Name, args.RequestingAssembly);
+                }
+            }
+
+            /// <summary>
+            /// Attempts to find and load an <see cref="Assembly"/> when the requesting <see cref="Assembly"/>
+            /// is unknown.
+            /// </summary>
+            /// <remarks>
+            /// In this case we simply look next to all the assemblies we have previously loaded for one with the
+            /// correct name and a matching <see cref="AssemblyIdentity"/>.
+            /// </remarks>
+            private static Assembly ResolveForUnknownRequestor(string requestedAssemblyName)
+            {
+                lock (s_guard)
+                {
+                    string requestedNameWithPolicyApplied = AppDomain.CurrentDomain.ApplyPolicy(requestedAssemblyName);
+
+                    Assembly assembly;
+                    if (s_assembliesFromNames.TryGetValue(requestedNameWithPolicyApplied, out assembly))
+                    {
+                        // We've already loaded an assembly by this name; use that.
```

Comment:
Ah never mind. Based on following line looks like it does include identify info...